src/App.tsx

```diff
@@ -1,17 +1,169 @@
+  const [noEnteredUser, setUserError] = useState(false);
+  const nextTodoId = updatedTodos.length + 1;
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTitleError(!title);
+
+    setUserError(!selectUserId);
+
+    if (title && selectUserId) {
+      const nextTodos = {
+        id: nextTodoId,
+        userId: parseInt(selectUserId, 10),
+        title,
+        completed: false,
+        user: users.find((user) => user.id === +selectUserId) || null,
+      };
+
+      setNewTodos([...updatedTodos, nextTodos]);
+      setTitle('');
+      setUserIdSelect('');
+    }
+  };
+
+  const getSelectedUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+    setUserIdSelect(event.target.value);
+    setUserError(false);
+  };
+
+  const getChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const getChangeCompleted = (event: React.ChangeEvent<HTMLInputElement>,
+    newTodo: Todo) => {
+    const { checked } = event.target;
+    const changeCompleted = updatedTodos.map((todo) => {
+      if (todo.id === newTodo.id) {
+        return {
+          ...todo,
+          completed: checked,
+        };
+      }
+
+      return todo;
+    });
+
+    setNewTodos([...changeCompleted]);
+  };
+
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
+    <div className="App container is-max-desktop">
+      <h1 className="title is-1 title-centered">Add todo form</h1>
+
+      <form
+        onSubmit={onSubmit}
+      >
+
+        <div className="field is-grouped is-grouped-centered">
+          <div className="control">
+            <div className="select is-primary">
+              <select
+                name="user"
+                value={selectUserId}
+                onChange={getSelectedUser}
+              >
+                <option value="" disabled>Select user</option>
+
+                {users.map(user => (
+
+                  <option
+                    value={user.id}
+                    key={user.id}
+                  >
+                    {user.name}
+                  </option>
+
+                ))}
+              </select>
+
+              {noEnteredUser
+                && <span className="help is-danger"> Field is required!</span>}
+            </div>
+          </div>
+          <div className="control">
+            <input
+              className="input is-info"
+              type="text"
+              placeholder="Title"
+              value={title}
+              onChange={getChangeTitle}
+            />
+            {noEnteredTitle
+              && <span className="help is-danger">Field is required!</span>}
+          </div>
+          <div className="control">
+            <button className="button is-success" type="submit">Submit</button>
+          </div>
+        </div>
+
+      </form>
+
+      <ul className="todo">
```

Comment:
It'll be better if you transfer component with todos in separate file.

Reply:
Ok, thanks. I agree with that )